tests/unit/test_config_provider.py

```diff
@@ -942,3 +943,172 @@ def test_resolve_auto_mode_imds_region_provider_connect_timeout(self):
             )
             mode = smart_defaults_factory.resolve_auto_mode('us-west-2')
             assert mode == 'standard'
+
+
+def _known_service_names_and_ids():
+    my_session = session.get_session()
+    loader = my_session.get_component('data_loader')
+    available_services = loader.list_available_services('service-2')
+
+    result = set()
+    for service_name in available_services:
+        model = my_session.get_service_model(service_name)
+        result.add((model.service_name, model.service_id))
+    return sorted(result)
```

Comment:
Why does this need to be sorted? If it isn't necessary we could get rid of the set and yield instead:
```python
yield (model.service_name, model.service_id)

Reply:
The tests fail with this error if they are not sorted:

https://github.com/pytest-dev/pytest-xdist/issues/432

Reply:
Hmm that says its specifically for running multiple tests in parallel. Are you still seeing that error without passing `-n`? I just find this very strange since we have many of these types of functions across our test suite where we don't need to sort.